### PR TITLE
Remove dependency on kotlin-reflect and improve proguard compatibility

### PR DESCRIPTION
**Pull request by rubengees**
> This PR is for the latest `4.0.0` RC.

### Remove dependency on `kotlin-reflect`

Plain JVM reflection is now used to instantiate the fonts and processors. Since the fonts are Kotlin objects, a special case for that is added. Instantiating the objects via the constructor works, but defeats the semantics of an object so I went with checking (defensively) for the `INSTANCE` field and getting the value of that.

### Update consumer-proguard-rules

- The consumer rule for community material was incorrect so it is updated.
- There were no rules for processors so these are added.

I verified all of this in one of my apps (debug and shrinked with latest R8) and also the sample. No warnings appear and everything works.

### CODE REVIEW

**Comment by mikepenz** [comment]
@rubengees that is a very great PR. thank you very much I hope I can review + test soon. 

Little bit busy as I am preparing for the Google IO

**Comment by rubengees** [comment]
Thanks :) Cool, no worries, I can just get the change from JitPack for the time being.

File: library-core/src/main/java/com/mikepenz/iconics/context/ReflectionUtils.kt [hunk 1]
```diff
@@ -59,11 +58,35 @@ internal object ReflectionUtils {
...
-    @Suppress("NOTHING_TO_INLINE")
-    inline fun <T : Any> getInstanceOf(cls: KClass<T>): T {
-        return cls.objectInstance ?: cls.createInstance()
+    /**
+     * A helper method to instantiate a class or Kotlin object. This could be easier by using
+     * kotlin-reflection, but is done with JVM reflection to avoid the large dependency.
+     */
+    @Suppress("NOTHING_TO_INLINE", "UNCHECKED_CAST")
+    inline fun <T : Any> getInstanceOf(cls: Class<T>): T {
+        val instanceField = try {
```
Comment (zTrap) [inline library-core/src/main/java/com/mikepenz/iconics/context/ReflectionUtils.kt]:
```suggestion
        val instanceField = runCatching { cls.getField("INSTANCE") }.getOrNull()
```

File: library-core/src/main/java/com/mikepenz/iconics/context/ReflectionUtils.kt [hunk 2]
```diff
@@ -59,11 +58,35 @@ internal object ReflectionUtils {
...
+            cls.getField("INSTANCE")
```
Comment (zTrap) [inline library-core/src/main/java/com/mikepenz/iconics/context/ReflectionUtils.kt]:
```suggestion
```

File: library-core/src/main/java/com/mikepenz/iconics/context/ReflectionUtils.kt [hunk 3]
```diff
@@ -59,11 +58,35 @@ internal object ReflectionUtils {
...
+        } catch (err: NoSuchFieldException) {
```
Comment (zTrap) [inline library-core/src/main/java/com/mikepenz/iconics/context/ReflectionUtils.kt]:
```suggestion
```

File: library-core/src/main/java/com/mikepenz/iconics/context/ReflectionUtils.kt [hunk 4]
```diff
@@ -59,11 +58,35 @@ internal object ReflectionUtils {
...
+            null
```
Comment (zTrap) [inline library-core/src/main/java/com/mikepenz/iconics/context/ReflectionUtils.kt]:
```suggestion
```

File: library-core/src/main/java/com/mikepenz/iconics/context/ReflectionUtils.kt [hunk 5]
```diff
@@ -59,11 +58,35 @@ internal object ReflectionUtils {
...
+        }
```
Comment (zTrap) [inline library-core/src/main/java/com/mikepenz/iconics/context/ReflectionUtils.kt]:
```suggestion
```

File: library-core/src/main/java/com/mikepenz/iconics/context/ReflectionUtils.kt [hunk 6]
```diff
@@ -59,11 +58,35 @@ internal object ReflectionUtils {
...
+        return if (
+            instanceField != null &&
+            Modifier.isFinal(instanceField.modifiers) &&
+            Modifier.isStatic(instanceField.modifiers)
+        ) {
+            // This is a Kotlin object.
+            instanceField.get(null) as T
+        } else {
+            // This is a regular class.
+            val constructor = cls.getDeclaredConstructor().apply {
```
Comment (zTrap) [inline library-core/src/main/java/com/mikepenz/iconics/context/ReflectionUtils.kt]:
```suggestion
            cls.newInstance()
```

File: library-core/src/main/java/com/mikepenz/iconics/context/ReflectionUtils.kt [hunk 7]
```diff
@@ -59,11 +58,35 @@ internal object ReflectionUtils {
...
+                isAccessible = true
+            }
```
Comment (zTrap) [inline library-core/src/main/java/com/mikepenz/iconics/context/ReflectionUtils.kt]:
```suggestion
```

File: library-core/src/main/java/com/mikepenz/iconics/context/ReflectionUtils.kt [hunk 8]
```diff
@@ -59,11 +58,35 @@ internal object ReflectionUtils {
...
+
```
Comment (zTrap) [inline library-core/src/main/java/com/mikepenz/iconics/context/ReflectionUtils.kt]:
```suggestion
```

File: library-core/src/main/java/com/mikepenz/iconics/context/ReflectionUtils.kt [hunk 9]
```diff
@@ -59,11 +58,35 @@ internal object ReflectionUtils {
...
+            constructor.newInstance()
```
Comment (zTrap) [inline library-core/src/main/java/com/mikepenz/iconics/context/ReflectionUtils.kt]:
```suggestion
```

**Comment by mikepenz** [comment]
I am sorry that this is not yet merged. Traveling keeps me really busy and is hard to make time for everything :/

**Pull request merged**